common/components/meta-list/template.njk

```diff
@@ -11,6 +11,15 @@
         <dd class="app-meta-list__value">
           {{ item.value.html | safe if item.value.html else item.value.text }}
         </dd>
+
+        {% set updateLink = item.link %}
+        {% if updateLink %}
```

Comment:
Also, the addition of this no longer feels content agnostic. The naming of the variables is binding it specifically to the link saying "Update" or "Change", whereas in the terms of a component, it should have more flexibility.

I think it would be cleaner for this component to support a similar pattern to the [GOV.UK Summary list component](https://design-system.service.gov.uk/components/summary-list/) where the component can have "actions" and the presenter sends in the text and href constructed from the URLs object.

Reply:
I think this pattern of "actions" would be nice to use actually:

```
...
actions: {
  items: [
    {
      href: "#",
      text: "Change",
      visuallyHiddenText: "name"
    }
  ]
}
...
```

Reply:
tldr; the meta-list component is already a specialist component with hard-coded fields in it.

That feels like premature optimisation? Do we know that the metalist requires multiple actions?  The following would do what you suggest, but it can only do it by subsuming information about classes and attributes which to me seems even yuckier. 

`common/components/meta-list/template.njk`

```njk
{%- macro _actionLink(action) %}
  <a class="govuk-link {%- if action.classes %} {{ action.classes }}{% endif %}" href="{{ action.href }}" {%- for attribute, value in action.attributes %} {{attribute}}="{{value}}"{% endfor %}>
    {{ action.html | safe if action.html else action.text }}
    {%- if action.visuallyHiddenText -%}
      <span class="govuk-visually-hidden"> {{ action.visuallyHiddenText }}</span>
    {% endif -%}
  </a>
{% endmacro -%}

<dl class="app-meta-list {% if params.classes -%} {{ params.classes }}{%- endif %}">
  {% for item in params.items %}
    {% if item.value.text or item.value.html %}
      <div class="app-meta-list__item {% if item.classes -%} {{ item.classes }}{%- endif %}">
        {% if item.key.html or item.key.text %}
          <dt class="app-meta-list__key">
            {{ item.key.html | safe if item.key.html else item.key.text }}
          </dt>
        {% endif %}

        <dd class="app-meta-list__value">
          {{ item.value.html | safe if item.value.html else item.value.text }}
        </dd>

        {% if item.actions.items.length %}
          <dd class="app-meta-list__actions {%- if item.actions.classes %} {{ item.actions.classes }}{% endif %}">
            {% if item.actions.items.length == 1 %}
              {{ _actionLink(item.actions.items[0]) | indent(12) | trim }}
            {% else %}
              <ul class="app-meta-list__actions-list">
                {% for action in item.actions.items %}
                  <li class="app-meta-list__actions-list-item">
                    {{ _actionLink(action) | indent(18) | trim }}
                  </li>
                {% endfor %}
              </ul>
            {% endif %}
          </dd>
        {% endif %}
      </div>
    {% endif %}
  {% endfor %}
</dl>
```

`common/presenters/move-to-meta-list-component.js`

```js
...
    additional_information: additionalInfo,
  } = {},
  additionalProps = {}
) {
  const destination = get(toLocation, 'title', 'Unknown')
  const destinationLabel =
    moveType === 'prison_recall'
      ? i18n.t('fields::move_type.items.prison_recall.label')
      : destination
  const additionalInformation = additionalInfo ? ` — ${additionalInfo}` : ''
  const items = [
    {
      key: {
        text: i18n.t('fields::from_location.short_label'),
      },
      value: {
        text: get(fromLocation, 'title'),
      },
    },
    {
      ...additionalProps.move,
      key: {
        text: i18n.t('fields::move_type.short_label'),
      },
      value: {
        text: destinationLabel + additionalInformation,
      },
    },
    {
      ...additionalProps.date,
      key: {
        text: i18n.t('fields::date_type.label'),
      },
      value: {
        text: setDateToDisplay({ date, dateFrom, dateTo }),
      },
    },
...
```
`app/move/controllers/view.js`

```js
...
  const updateLinks = getUpdateLinks(updateSteps, req.t, updateUrls)

  const urls = {
    update: updateUrls,
  }

  const getSummaryItemProps = actions => {
    if (!actions) {
      return
    }
    return {
      classes: 'app-!-position-relative',
      actions: {
        classes: 'app-!-position-top-right govuk-!-font-size-16',
        items: [actions],
      },
    }
  }

  const metaListProps = {
    date: getSummaryItemProps(updateLinks.date),
    move: getSummaryItemProps(updateLinks.move),
  }

  const locals = {
    moveSummary: presenters.moveToMetaListComponent(move, metaListProps),
    personalDetailsSummary: presenters.personToSummaryListComponent(person),
...
```

`app/move/controllers/view/view.update.links.js`
```js
...
      updateLinks[category] = {
        category,
        attributes: {
          'data-update-link': category,
        },
        href: urls[category],
        html: t('moves::update_link.link_text', {
          context: 'with_category',
          category: categoryText,
        }),
      }
...
```

Can we leave this for more discussion and not block getting this in for now?

Reply:
I hear you about we might not need multiple at this stage so that may be overkill. My point was more around this component being more agnostic to the content of the links you're currently adding.

Like with the GOV.UK example, I think the difference between a component and using elements within a template or specific view is that a component should be more generic in its use. The name of things suggests that this would only ever be used to add an "Update" link, but in reality the component doesn't care what the link says.

I also understand we don't want to hold changes up, but I think changes to components just need to be thought of a bit more to not tie them down to single use, otherwise they lost the ability to be as easily re-used.

Reply:
The change needed could also be handled by using `item.value.html` instead of `item.value.text` and adding the link in the presenter.

And wouldn't require any updates to the component.

Reply:
Updated as discussed with meta-list component gaining action and not requiring app-!-style overrides